consensus/XDPoS/engines/engine_v2/engine.go

```diff
@@ -324,6 +324,29 @@ func (x *XDPoS_v2) YourTurn(chain consensus.ChainReader, parent *types.Header, s
 	return len(masternodes), preIndex, curIndex, false, nil
 }
 
+func (x *XDPoS_v2) IsAuthorisedAddress(header *types.Header, chain consensus.ChainReader, address common.Address) bool {
+	var extraField utils.ExtraFields_v2
+	err := utils.DecodeBytesExtraFields(header.Extra, &extraField)
+	if err != nil {
+		log.Error("[IsAuthorisedAddress] Fail to decode v2 extra data", "Hash", header.Hash())
+		return false
+	}
+	blockRound := extraField.Round
+
+	masterNodes := x.GetMasternodes(chain, header)
+
+	if len(masterNodes) == 0 {
+		log.Error("[IsAuthorisedAddress] Fail to find any master nodes from current block round epoch", "Hash", header.Hash(), "Round", blockRound, "MN", masterNodes)
```

Comment:
better have header.Number, and doesn't need to print "MN", masterNodes since len is 0